src/main/kotlin/uk/gov/justice/digital/hmpps/courtregister/resource/CourtMaintenanceResource.kt

```diff
@@ -0,0 +1,111 @@
+import io.swagger.v3.oas.annotations.responses.ApiResponse
+import io.swagger.v3.oas.annotations.security.SecurityRequirement
+import org.springframework.http.MediaType
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.validation.annotation.Validated
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.PutMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.courtregister.ErrorResponse
+import uk.gov.justice.digital.hmpps.courtregister.service.CourtService
+import javax.validation.constraints.NotBlank
+import javax.validation.constraints.Size
+
+@RestController
+@Validated
+@RequestMapping("/court-maintenance", produces = [MediaType.APPLICATION_JSON_VALUE])
+class CourtMaintenanceResource(private val courtService: CourtService) {
+  @PreAuthorize("hasRole('ROLE_MAINTAIN_REF_DATA')")
+  @Operation(
+    summary = "Update specified court details",
+    description = "Updates court information, role required is MAINTAIN_REF_DATA",
+    security = [ SecurityRequirement(name = "MAINTAIN_REF_DATA", scopes = ["write"])],
+    requestBody = io.swagger.v3.oas.annotations.parameters.RequestBody(content = [Content(mediaType = "application/json", schema = Schema(implementation = UpdateCourtDto::class))]),
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Court Information Updated",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = CourtDto::class))]
+      ),
+      ApiResponse(
+        responseCode = "400",
+        description = "Information request to update court",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Incorrect permissions to make court update"
+      ),
+      ApiResponse(
+        responseCode = "404",
+        description = "Court ID not found",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
+      )
+    ]
+  )
+  @PutMapping("/id/{courtId}")
+  fun updateCourt(
+    @Schema(description = "Court ID", example = "ACCRYC", required = true)
+    @PathVariable @Size(max = 12, min = 2, message = "Court ID must be between 2 and 12") courtId: String,
+    @RequestBody courtUpdateRecord: UpdateCourtDto
+  ): CourtDto =
+    courtService.updateCourt(courtId, courtUpdateRecord)
+
+  @PreAuthorize("hasRole('ROLE_MAINTAIN_REF_DATA')")
+  @PostMapping("")
+  @Operation(
+    summary = "Add a new court",
+    description = "Adds a new court information, role required is MAINTAIN_REF_DATA",
```

Comment:
+1 for having role information in the docs. 

It's a shame this can't be specified as a separate part of the swagger doc but this beats digging aound in the code base

Reply:
It is indeed.  I've taken to putting it on the description of a 403 response.